templates/statefulset.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.name}}
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Release.Name}}
+  serviceName: {{ .Values.service}}-headless

```

Comment:
drop -headless and just use the .Release.Name.  This will make it easier to manage when all of the types use the same name